src/debug/FuncTaskProvider.ts

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken, ShellExecution, Task, TaskProvider, workspace, WorkspaceFolder } from 'vscode';
+import { isFunctionProject } from '../commands/createNewProject/isFunctionProject';
+import { extInstallCommand, func, funcExtInstallCommand, funcHostStartCommand, funcWatchProblemMatcher, hostStartCommand, ProjectLanguage, projectLanguageSetting } from '../constants';
+import { getFuncExtensionSetting } from '../ProjectSettings';
+import { FuncDebugProviderBase } from './FuncDebugProviderBase';
+import { JavaDebugProvider } from './JavaDebugProvider';
+import { NodeDebugProvider } from './NodeDebugProvider';
+import { PythonDebugProvider } from './PythonDebugProvider';
+
+export class FuncTaskProvider implements TaskProvider {
+    private readonly _nodeDebugProvider: NodeDebugProvider;
+    private readonly _pythonDebugProvider: PythonDebugProvider;
+    private readonly _javaDebugProvider: JavaDebugProvider;
+
+    constructor(nodeDebugProvider: NodeDebugProvider, pythonDebugProvider: PythonDebugProvider, javaDebugProvider: JavaDebugProvider) {
```

Comment:
Why have this be part of the constructor rather than just export it from the respective files?  It doesn't seem like there are any dynamic properties in the constructors for any of the DebugProviders.